ddtrace/contrib/cherrypy/middleware.py

```diff
@@ -0,0 +1,114 @@
+"""
+Datadog trace code for cherrypy.
+"""
+
+# stdlib
+import logging
+
+# project
+from ddtrace import config
+from .. import trace_utils
+from ... import compat
+from ...ext import errors, SpanTypes
+from ...propagation.http import HTTPPropagator
+
+# 3p
+import cherrypy
+from cherrypy.lib.httputil import valid_status
+
+log = logging.getLogger(__name__)
+
+SPAN_NAME = "cherrypy.request"
+
+
+class TraceTool(cherrypy.Tool):
+    def __init__(self, app, tracer, service, use_distributed_tracing):
+        self.app = app
+        self._tracer = tracer
+        self._service = service
+        self._use_distributed_tracing = use_distributed_tracing
+
+        cherrypy.Tool.__init__(self, "on_start_resource", self._on_start_resource, priority=95)
+
+    def _setup(self):
+        cherrypy.Tool._setup(self)
+        cherrypy.request.hooks.attach("on_end_request", self._on_end_request, priority=5)
+        cherrypy.request.hooks.attach("after_error_response", self._after_error_response, priority=5)
+
+    def _on_start_resource(self):
+        if self._use_distributed_tracing:
```

Comment:
```suggestion
        if config.cherrypy.distributed_tracing or self._use_distributed_tracing:
```

(once the global config entry is added)

Reply:
Added with protections if the config object is manipulated elsewhere